### ISSUE TITLE
Better presentation of wind field

### ISSUE THREAD

**Issue by leeyupeng119** [GenericMappingTools/gmt#6108]
**Description of the desired feature**
I use **grdvector** function to plot the wind field, I adjusted -Q and -Si parameters repeatedly, but I couldn't get satisfactory results.  

Here is my final code (turn parameters again and again) and output figure,
```
gmt begin wind2_gmt.png
gmt coast  -JM5i -R75/140/15/55 -Baf -A4000  -W0.25p,black
gmt grdvector u.nc v.nc -Ix5 -Q4p+e+h0.1+n1k -Gblack -W0.5p -Si30k
gmt end
```
![GMT plot](https://user-images.githubusercontent.com/55687784/145677155-f8e8ebc8-bd3d-4852-b793-faf222479381.png)

And I did the same thing with NCL (NCAR Command Language), which is a very commonly used software in atmospheric science research. And here's what NCL output, 
![NCL plot](https://user-images.githubusercontent.com/55687784/145677111-cec694bc-7481-451d-89ff-b8f2240f4251.png)
By comparing the two plots, it can be found that GMT still has some limitations in drawing wind fields, the main limitations are as follows：

1.  Wind direction is not easily discernible in the weak wind speed zone, so you can't see the circulation pattern very well.
2.  There is no feature for automatically generating the legend of the reference vector.

**Are you willing to help implement and maintain this feature?** Yes

In my opinion, when the wind speed is very small, the stem may disappear, but there should be a certain lower limit for the reduction of the arrow, and the overall shape of the arrow should not be destroyed, so that the wind direction of each grid point can be clearly identified, which is very important in circulation analysis. In the research of meteorology,  we pay attention not only to the value of the wind speed but also to the wind direction. Therefore, would you consider adding the features I mentioned above? Thanks in advance.

Maybe I am not good at using **grdvector**, so I have uploaded the data, welcome you to test and turning.
[data_uv.zip](https://github.com/GenericMappingTools/gmt/files/7697118/data_uv.zip)

**Comment by PaulWessel** [comment]
Thanks for this example and the comparison. Having **+n** be subjected to a minimum shrinking is a good idea that can be implemented.  I will play with your data example to see how we can improve the plot. And adding -l for auto-legend is on my mental agenda anyway.

**Comment by PaulWessel** [comment]
So it looks like the vector head never shrinks in the NCL plot but you could do the same in GMT, no (not using +n)?  Aren't you comparing apples and oranges a bit here?  Anyway, some possibilities would be to add an optional minimum length to +n so syntax would be **+n**_start_[/_stop_], so shrinking will begin for vectors shorter than _start_ but only down to _stop_ [0].

If you want open vector heads like the NCL then you use **+eA** to change what head you want.  So although I have not tried yet, seems like you could make a map very similar to the NCL with GMT if you just select no shrinking and pick the plain vector head.

**Comment by joa-quim** [comment]
NCL vector heads also shrink, but apparently not more than  3 to 4 times.

**Comment by PaulWessel** [comment]
Yes you are right, there are clear differences but where we go to zero they stop shrinking.  I agree going to zero makes it impossible to see directions (you just get a point if you are lucky), so we can consider improvements here.

**Comment by PaulWessel** [comment]
Maybe the syntax could be **+n**_norm_[/_frac_] where default _frac_ = 0 but if given indicates how much smaller than the given vector head size we shall shrink.  E.g., a _frac_ of 0.5 means just shrink down to half the size and no further.

**Comment by joa-quim** [comment]
I would even prefer that default *frac* would be something that still allowed seeing the vector head. This would be a small breaking but we can always call the trick of saying previous behavior was a bug.

**Comment by PaulWessel** [comment]
We can discuss that.  I am opening a WIP branch PR that implements the /frac and have an example with some issues to comment on.

**Comment by PaulWessel** [comment]
After fixing the plain geovector and the minimum shrink value, I get stuff like this which is approximating what the NCL does if you prefer such vectors.

![wind2_gmt](https://user-images.githubusercontent.com/26473567/145883166-ff335014-f940-403d-9bdc-73fa2333781f.png)

**Comment by PaulWessel** [comment]
Hopefully, @leeyupeng119 can play with the revised vector head via **+eA** and the terminal shrink threshold via **+n**1k/0.7 (I used 0.7 in the above plot) and give more feedback.

**Comment by leeyupeng119** [comment]
@PaulWessel, It looks awesome and I think this would work well for my needs. Thank you all.

**Comment by leeyupeng119** [comment]
I've downloaded the code of master brunch and compiled it. It works very well for values of the new parameter from 0.7 to 0.9 with the +eA option. But when there is no A, some vector in the weak wind area is a bit strange. The stems ran to the front of the arrows instead of the tail.
Here is the code and a screenshot of strange zones.

```
 gmt begin wind2_gmt pdf
 gmt coast  -JM5i -R75/140/15/55 -Baf -A4000  -W0.25p,black
 gmt grdvector u.nc v.nc -Ix5 -Q4p+e+n1k/.9 -Gblack -W0.5p -Si20k
 gmt end
```

![image](https://user-images.githubusercontent.com/55687784/145921840-ac64b3c7-6e00-4b34-b654-5342fe2517fa.png)

**Comment by PaulWessel** [comment]
Thanks, I noticed that too.  Will have a look.

**Comment by gd-a** [comment]
> And adding -l for auto-legend is on my mental agenda 

Do you plan on adding an option to define at which latitude the legend takes reference from? Or simply in the middle of the plot ? (I'm more comfortable with the latter)

Most importantly, will the scaling be applied accordingly from an user-defined value or from a mean ? (I'm more comfortable with the former)

Thanks @PaulWessel !

**Comment by meghanrjones** [comment]
Final issues were fixed by #6161. @leeyupeng119, please comment if you notice any other issues.
